docking_server:
  ros__parameters:
    # ---------------------------------------- STATE MACHINE PARAMS ----------------------------------
    # -------------------------------------------- GLOBAL PARAMS --------------------------------------
    timeout: 3                  # timeout for blocking events
    debug: true                   # publish tf's with debug poses
    epsilon: 0.1                  # epsilon added to calculation of finishing condition of movement loops
    docking_point_distance: 0.8   # distance of the docking point from docking base
    battery_diff: 0.2             # difference in battery voltage to consider charging the rover
    max_battery_average: 11.0     # maximum battery voltage average at the start of the docking, to consider charging as one of the stop conditions
    battery_averaging_time: 1.0   # time over which the average battery voltage will be calculated
    effort_threshold: 2.0         # summarized effort threshold to consider the rover moving in place - getting blocked on the base
    effort_buffer_size: 10        # size of wheel motor efforts buffer - average of the sums is considered in comparison with effort threshold
    motor_cd_time: 2.0            # time for effort on motors to drop down

    # -------------------------------------------- START STATE PARAMS ---------------------------------
    # start_state/timeout: 3.0  # specific timeout for "StartState" state blocking events


    # -------------------------------------------- CHECK AREA PARAMS ----------------------------------
    check_area/threshold_angle: 0.17      # half of angle specifying docking area (in radians)
    check_area/docking_distance: 2.0      # distance of the target point in docking area from the docking base
    # check_area/timeout: 3.0  # specific timeout for "CheckArea" state blocking events


    # ---------------------------------------- ROTATE TO DOCK AREA PARAMS -----------------------------
    rotate_to_dock_area/speed_min: 0.1   # minimal angular speed
    rotate_to_dock_area/speed_max: 0.4   # maximum angular speed
    rotate_to_dock_area/angle_min: 0.05  # the angle (in radians) left for which the rover will move with the minimal speed
    rotate_to_dock_area/angle_max: 1.05  # the angle (in radians) left for which the rover will move with the maximum speed
    # rotate_to_dock_area/epsilon: 0.1   # specific epsilon for "RotateToDockArea" state calculations
    # rotate_to_dock_area/timeout: 3.0   # specific timeout for "RotateToDockArea" state blocking events


    # ---------------------------------------- RIDE TO DOCK AREA PARAMS -------------------------------
    ride_to_dock_area/speed_min: 0.05    # minimal linear speed
    ride_to_dock_area/speed_max: 0.4     # maximum linear speed
    ride_to_dock_area/distance_min: 0.1  # distance left (in meters) for which the rover will move with the minimal speed
    ride_to_dock_area/distance_max: 0.5  # distance left (in meters) for which the rover will move with the maximum speed
    # ride_to_dock_area/epsilon: 0.1     # specific epsilon for "RideToDockArea" state calculations
    # ride_to_dock_area/timeout: 3.0     # specific timeout for "RideToDockArea" state blocking events


    # ------------------------------------------ ROTATE TO BOARD PARAMS ------------------------------
    rotate_to_board/speed_min: 0.1     # minimal angular speed
    rotate_to_board/speed_max: 0.4     # maximum angular speed
    rotate_to_board/angle_min: 0.05    # the angle (in radians) left for which the rover will move with the minimal speed
    rotate_to_board/angle_max: 1.05    # the angle (in radians) left for which the rover will move with the maximum speed
    # rotate_to_board/epsilon: 0.1     # specific epsilon for "RotateToBoard" state calculations
    # rotate_to_board/timeout: 3.0     # specific timeout for "RotateToDockBoard" state blocking events


    # ------------------------------------- ROTATE TO DOCKING POINT PARAMS ----------------------------
    rotate_to_docking_point/speed_min:  0.1                     # minimal angular speed
    rotate_to_docking_point/speed_max:  0.4                     # maximum angular speed
    rotate_to_docking_point/angle_min:  0.05                    # the angle (in radians) left for which the rover will move with the minimal speed
    rotate_to_docking_point/angle_max:  1.0                     # the angle (in radians) left for which the rover will move with the maximum speed
    rotate_to_docking_point/min_docking_point_distance: 0.1     # minimum distance to the docking point for the rover to execute this state (prevents from infinite execution)
    # rotate_to_docking_point/epsilon: 0.1                      # specific epsilon for "RotateToDockingPoint" state calculations
    # rotate_to_docking_point/timeout: 3.0                      # specific timeout for "RotateToDockingPoint" state blocking events


    # -------------------------------------- REACHING DOCKING POINT PARAMS ----------------------------
    reach_docking_point/speed_min: 0.1          # minimal linear speed
    reach_docking_point/speed_max: 0.3          # maximum linear speed
    reach_docking_point/distance_min: 0.05      # distance left (in meters) for which the rover will move with the minimal speed
    reach_docking_point/distance_max: 1.5       # distance left (in meters) for which the rover will move with the maximum speed
    reach_docking_point/bias_speed_min: 0.05    # minimal angular speed for the correction movement
    reach_docking_point/bias_speed_max: 0.3     # maximum angular speed for the correction movement
    reach_docking_point/bias_min: 0.01          # the angle (in radians) to correct for which the rover will rotate with the minimal speed
    reach_docking_point/bias_max: 0.1           # the angle (in radians) to correct for which the rover will rotate with the maximum speed
    # reach_docking_point/epsilon: 0.1          # specific epsilon for "ReachingDockingPoint" state calculations
    # reach_docking_point/timeout: 3.0          # specific timeout for "ReachingDockingPoint" state blocking events


    # ------------------------------------- REACHING DOCKING ORIENTATION PARAMS -----------------------
    reach_docking_orientation/speed_min: 0.1     # minimal angular speed
    reach_docking_orientation/speed_max: 0.4     # maximum angular speed
    reach_docking_orientation/angle_min: 0.05    # the angle (in radians) left for which the rover will move with the minimal speed
    reach_docking_orientation/angle_max: 1.0     # the angle (in radians) left for which the rover will move with the maximum speed
    # reaching_docking_orientation/epsilon: 0.1     # specific epsilon for "ReachingDockingOrientation" state calculations
    # reaching_docking_orientation/timeout: 3.0     # specific timeout for "ReachingDockingOrientation" state blocking events


    #  -------------------------------------------------- DOCKING ROVER -------------------------------
    dock/speed_min: 0.05        # minimal linear speed
    dock/speed_max: 0.2         # maximum linear speed
    dock/distance_min: 0.05     # distance left (in meters) for which the rover will move with the minimal speed
    dock/distance_max: 0.8      # distance left (in meters) for which the rover will move with the maximum speed
    dock/epsilon: 0.25          # specific epsilon for "DockingRover" state calculations
    dock/bias_speed_min: 0.05    # minimal angular speed for the correction movement
    dock/bias_speed_max: 0.3     # maximum angular speed for the correction movement
    dock/bias_min: 0.01          # the angle (in radians) to correct for which the rover will rotate with the minimal speed
    dock/bias_max: 0.1           # the angle (in radians) to correct for which the rover will rotate with the maximum speed
    # dock/timeout: 3.0         # specific timeout for "DockingRover" state blocking events
